### DISCUSSION TITLE
Do anyone have a detailed tutorial on publishing aab files?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2929] [Q&A] [answered]
I just searched for some information on the website and couldn't find a detailed tutorial on publishing aab.
I have tried before, but it was not successful because Steve's SDK was not installed.
I have been using Animate to publish APK for use now, and I am worried that this APK will be phased out in the future, so I want to try using AAB format. If this concern is unnecessary, there is no need to try again. Because my application is currently only being tested on a single machine.
thanks.

**Comment by raresn** [comment]
Animate is a great tool, but have you considered using command line and adb directly? You can just publish your swf from Animate, then don't worry anymore about Adobe or anyone else updating their interface so you are able to compile. 
A command line for compiling aab would be something like this:
AdobeAIRSDK\bin\adt.bat -package -target aab -storetype pkcs12 -keystore androidMarket.p12 -storepass xxx "yourApp.aab" "yourApp.xml" "yourApp.swf" foldersAndFilesHereOneByOneSeparatedBySpace -extdir extensions

**Reply by Jason0247** [reply]
Can the aab format only be published through the command line?
I see many people using the command line. What are the benefits or advantages of it compared to the release of Animate?

**Reply by 2jfw** [reply]
>Can the aab format only be published through the command line?
I see many people using the command line. What are the benefits or advantages of it compared to the release of Animate?

[Site Note on this^^ currently being "Marked as answer": I would think that [this](https://github.com/airsdk/Adobe-Runtime-Support/discussions/2929#discussioncomment-7615063) is the actual answer to your question rather than this one ;) ]

**Reply by Jason0247** [reply]
You're right, I think I accidentally clicked on this button.
Thank you for the reminder.

**Comment by FliplineStudios** [comment, marked as answer]
@Jason0247 There's a tutorial for packaging AAB on the AIR SDK Dev site here:
https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles

There's also a section called "IDE's without AAB option", which I believe you would need to check out if you're using Animate to generate the aab.  Animate wasn't originally designed to package aab, so there's a bit of a workaround there.

Personally I've always packaged anything AIR-related from the command line, whether I'm generating the SWF it's based on from Animate or elsewhere like FlashDevelop.  In our case we build separate SWFs for iOS and Android, and package them up with other files like extensions and icons and storyboards, so it's just easier to have them all in separate folders, and run a command line script for each for the final packaging.  And this way I can just download new AIR SDKs as they're available, and re-package things with them from the command line without having to rebuild the source SWF used (as long as I don't want to use any new APIs).

**Reply by Jason0247** [reply]
Thank you very much. I'll go and learn more first.
I have been using the packaging interface of the Animate tool for packaging before, and I am not very familiar with this knowledge.
